src/host/main.cc

```diff
@@ -110,13 +322,53 @@ int main()
 			) == GLFW_PRESS;
 		}
 
+#if defined(DEVELOPMENT_MODE)
+		if (poll_src_changes || input.down(GRAVE_ACCENT)) {
+			if (file_modification_time_try_to_update(src_name, &game.prev_load_time)) {
+				puts("reloading library");
+				if (GameLibrary_reload(game) && game.on_reload != nullptr) {
+					game.on_reload(&state, &renderer, &user_data);
+				}
+			}
+		}
+
+#if defined(DEVELOPMENT_MODE) && defined(BURNER_EXPERIMENTAL) // TODO put into single function
+		// early save states testing, not 100% sure whether I covered all cases for time "travel"
+		if (input.down(SLASH)) {
+
+			#define ANSI_COLOR_RED     "\x1b[31m"
+			#define ANSI_COLOR_GREEN   "\x1b[32m"
+			#define ANSI_COLOR_YELLOW  "\x1b[33m"
+			#define ANSI_COLOR_BLUE    "\x1b[34m"
+			#define ANSI_COLOR_MAGENTA "\x1b[35m"
+			#define ANSI_COLOR_CYAN    "\x1b[36m"
+			#define ANSI_COLOR_RESET   "\x1b[0m"
+
+			memcpy(saves.data, (char*)user_data, user_data_size);
+			saves.input_states[0] = input;
+			saves.time_states[0] = time;
+
+			printf(ANSI_COLOR_GREEN "State saved, [slot=%d, time=%f]" ANSI_COLOR_RESET "\n", 
+				0, time);
+
+		} else if (input.down(PERIOD)) {
+			memcpy(user_data, saves.data, user_data_size);
+			time_offset = glfwGetTime() - saves.time_states[0];
```

Comment:
I think glfw lets you set the time yourself, come to think of it, so I can avoid these offset calculations. SDL does not have such a function, so that's probably why I did things this way.